test-cargo-miri/src/lib.rs

```diff
@@ -1,13 +1,29 @@
+extern crate exported_symbol;
+
 /// Doc-test test
 /// ```rust
 /// assert!(cargo_miri_test::make_true());
+/// // Repeat calls to make sure the `Instance` cache is not broken.
+/// for _ in 0..3 {
+///     extern "Rust" {
+///         fn exported_symbol() -> i32;
+///         fn make_true() -> bool;
+///     }
+///     assert_eq!(unsafe { exported_symbol() }, 123456);
```

Comment:
What is the reason why you are specifically testing this in a doctesst?

Reply:
Not for a specific reason. I just found that writing it here is convenient.

Reply:
I think I'd prefer if these would be regular unit tests. That will also ensure they are run in cross-compilation mode.